packages/core/data-transfer/lib/providers/local-strapi-destination-provider/index.ts

```diff
@@ -0,0 +1,138 @@
+// import { createLogger } from '@strapi/logger';
+import type { IDestinationProvider, IMetadata, ProviderType } from '../../../types';
+import { deleteAllRecords, DeleteOptions } from './restore';
+
+import chalk from 'chalk';
+import { Writable } from 'stream';
+
+import { mapSchemasValues } from '../../utils';
+
+export const VALID_STRATEGIES = ['restore', 'merge'];
+
+interface ILocalStrapiDestinationProviderOptions {
+  getStrapi(): Strapi.Strapi | Promise<Strapi.Strapi>;
+  restore?: DeleteOptions;
+  strategy: 'restore' | 'merge';
+}
+
+// TODO: getting some type errors with @strapi/logger that need to be resolved first
+// const log = createLogger();
+const log = console;
+
+export const createLocalStrapiDestinationProvider = (
+  options: ILocalStrapiDestinationProviderOptions
+) => {
+  return new LocalStrapiDestinationProvider(options);
+};
+
+class LocalStrapiDestinationProvider implements IDestinationProvider {
+  name: string = 'destination::local-strapi';
+  type: ProviderType = 'destination';
+
+  options: ILocalStrapiDestinationProviderOptions;
+  strapi?: Strapi.Strapi;
+
+  constructor(options: ILocalStrapiDestinationProviderOptions) {
+    this.options = options;
+  }
+
+  async bootstrap(): Promise<void> {
+    this.#validateOptions();
+    this.strapi = await this.options.getStrapi();
+  }
+
+  async close(): Promise<void> {
+    await this.strapi?.destroy?.();
+  }
+
+  #validateOptions() {
+    if (!VALID_STRATEGIES.includes(this.options.strategy)) {
+      throw new Error('Invalid stategy ' + this.options.strategy);
+    }
+  }
+
+  async #deleteAll() {
+    if (!this.strapi) {
+      throw new Error('Strapi instance not found');
+    }
+    return await deleteAllRecords(this.strapi, this.options.restore);
+  }
+
+  async beforeStreaming() {
```

Comment:
Again, I think I prefer a "before transfer" event instead of "before streaming".
My view on this is that we're doing operations that need to be run before we start the transfer processes, which has -in theory- nothing to do with having to do this before streaming anything.
What do you think?

Reply:
To be clearer, I think that here we're talking about actions that are run sequentially in a transfer process (before, transfer, after), and not a streaming one (even though under the hood we're using streams).